### PR TITLE
.*: Remove `DAGQueryBlock`

### PR DESCRIPTION
**Pull request by SeaRise**
### What problem does this PR solve?

Issue Number: ref #8276

Problem Summary:

### What is changed and how it works?

- remove `DAGQueryBlock` related codes
- remove `PlanQuerySource`
- minor refine for `executeQuery`

```commit-message

```

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Interpreters/Settings.h [hunk 1]
```diff
@@ -313,7 +313,6 @@ struct Settings
     M(SettingUInt64, max_cached_data_bytes_in_spiller, 1024ULL * 1024 * 20, "Max cached data bytes in spiller before spilling, 20 MB as the default value, 0 means no limit")                                                           \
     M(SettingUInt64, max_spilled_rows_per_file, 200000, "Max spilled data rows per spill file, 200000 as the default value, 0 means no limit.")                                                                                         \
     M(SettingUInt64, max_spilled_bytes_per_file, 0, "Max spilled data bytes per spill file, 0 as the default value, 0 means no limit.")                                                                                                 \
-    M(SettingBool, enable_planner, true, "Enable planner")                                                                                                                                                                              \
```
Comment (JaySon-Huang) [inline dbms/src/Interpreters/Settings.h:316]: Removing this config is a breaking for backward compatibility, if some user explicitly set `profiles.default.enable_planner` in the config file, then they are not able to start tiflash successfully after this line is removed.

We need to inform the user in the docs of next tidb release note

Comment (SeaRise) [reply dbms/src/Interpreters/Settings.h]: https://github.com/pingcap/tiflash/pull/9105/files#diff-955a52760602455c12a0bfdf2ead25c1b536ed4264ff0d5d0e11e933693a0363R182-R188
Restored setting and printed an info log.

Comment (JaySon-Huang) [reply dbms/src/Interpreters/Settings.h]: lgtm

**Review by Lloyd-Pottiger — APPROVED** [review]

**Review by windtalker — APPROVED** [review]
LGTM

**Pull request merged**